src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -102,26 +104,29 @@ def _webhook_event_data(
 @instrumented_task(name="sentry.sentry_apps.tasks.sentry_apps.send_alert_event", **TASK_OPTIONS)
 @retry_decorator
 def send_alert_event(
-    event: Event | GroupEvent,
```

Comment:
You can't remove this parameter entirely. There will be messages in the queues that have this parameter. Instead, I would make the signature:

```python
def send_alert_event(
    rule: str,
    sentry_app_id: int,
    # instance_id, group_id, occurence_id will become required
    instance_id: str | None = None
    group_id: int | None = None,
    occurence_id: str | None = None,
    event: Event | GroupEvent | None = None,   # deprecated
    additional_payload_key: str | None = None,
    additional_payload: Mapping[str, Any] | None = None,
) -> None:
```

Another approach would be to add `**kwargs` to the signature and read the deprecated `event` and new parameters from `kwargs`.

```python
def send_alert_event(
    rule: str,
    sentry_app_id: int,
    additional_payload_key: str | None = None,
    additional_payload: Mapping[str, Any] | None = None,
    **kwargs: Any
) -> None:
```

Reply:
We talked about this a bit but new plan is to do cutover, i.e
1. Make a new task (send_alert_event_v2), put the task behind option flag and wait for the task to be present in all regions
2. Turn on the option flag and observe results/performance in regions
3. If all goes well, move option to 100
4. Lastly, deprecate & remove the old task
5. profit 🥳 
Lmk if I missed any steps 😓

Reply:
That sounds like a good plan to me.